ingestion/src/metadata/ingestion/source/database/domodatabase.py

```diff
@@ -0,0 +1,199 @@
+from metadata.ingestion.models.ometa_tag_category import OMetaTagAndCategory
+from metadata.ingestion.ometa.ometa_api import OpenMetadata
+from metadata.ingestion.source.database.database_service import (
+    DatabaseServiceSource,
+    SQLSourceStatus,
+)
+from metadata.utils import fqn
+from metadata.utils.connections import get_connection, test_connection
+from metadata.utils.filters import filter_by_table
+from metadata.utils.logger import ingestion_logger
+
+logger = ingestion_logger()
+
+
+class DomodatabaseSource(DatabaseServiceSource):
+    """
+    Implements the necessary methods to extract
+    Database metadata from Domo Database Source
+    """
+
+    def __init__(self, config: WorkflowSource, metadata_config: OpenMetadataConnection):
+        self.config = config
+        self.source_config: DatabaseServiceMetadataPipeline = (
+            self.config.sourceConfig.config
+        )
+        self.metadata = OpenMetadata(metadata_config)
+        self.service_connection = self.config.serviceConnection.__root__.config
+        self.status = SQLSourceStatus()
+        self.connection = get_connection(self.service_connection)
+        self.domo_client = self.connection.client
+        self.client = DomoClient(self.service_connection)
+        super().__init__()
+
+    @classmethod
+    def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
+        config = WorkflowSource.parse_obj(config_dict)
+        connection: DomoDatabaseConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, DomoDatabaseConnection):
+            raise InvalidSourceException(
+                f"Expected DomoDatabaseConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_database_names(self) -> Iterable[str]:
+        database_name = "default"
+        yield database_name
+
+    def yield_database(self, database_name: str) -> Iterable[CreateDatabaseRequest]:
+        yield CreateDatabaseRequest(
+            name=database_name,
+            service=EntityReference(
+                id=self.context.database_service.id,
+                type="databaseService",
+            ),
+        )
+
+    def get_database_schema_names(self) -> Iterable[str]:
+        scheme_name = "default"
+        yield scheme_name
+
+    def yield_database_schema(
+        self, schema_name: str
+    ) -> Iterable[CreateDatabaseSchemaRequest]:
+        yield CreateDatabaseSchemaRequest(
+            name=schema_name,
+            database=EntityReference(id=self.context.database.id, type="database"),
+        )
+
+    def get_tables_name_and_type(self) -> Optional[Iterable[Tuple[str, str]]]:
+        schema_name = self.context.database_schema.name.__root__
+        table_name = ""
+        try:
+            tables = list(self.domo_client.datasets.list())
+            for table in tables:
+                table_name = table["id"]
```

Comment:
let's rename this variable as table_id otherwise it might be misleading

Reply:
Yes , sure.